### PR TITLE
Add automated acceptance tests for cmd_exec API

### PR DESCRIPTION
**Pull request by cgranleese-r7**
This PR adds acceptance tests for the new `cmd_exec` API changes. More information on these changes can be found in the following PRs:
- metasploit-framework: https://github.com/rapid7/metasploit-framework/pull/19108 
- metasploit-payloads: https://github.com/rapid7/metasploit-payloads/pull/701
- mettle: https://github.com/rapid7/mettle/pull/258

## Verification

- [ ] Code changes are sane
- [ ] CI passes

### CODE REVIEW

File: .github/workflows/meterpreter_acceptance.yml
Comment (adfoster-r7) [inline .github/workflows/meterpreter_acceptance.yml:1]: Can we put up a separate PR to rename this ahead of time, it'll help clean up this PR a lot :+1:

Comment (cgranleese-r7) [reply .github/workflows/meterpreter_acceptance.yml:1]: Sure

Comment (cgranleese-r7) [reply .github/workflows/meterpreter_acceptance.yml:1]: https://github.com/rapid7/metasploit-framework/pull/19423

File: data/cmd_exec/show_args
Comment (adfoster-r7) [inline data/cmd_exec/show_args:1]: Let's double check these names are legit :+1:

File: spec/support/acceptance/command_shell.rb [hunk 1]
```diff
@@ -0,0 +1,102 @@
+module Acceptance::CommandShell
```
Comment (adfoster-r7) [inline spec/support/acceptance/command_shell.rb:1]: Instead of the copy/pasta here, is it not possible to just rename the existing file and reuse it for both meterpreter and command shells? 👀

Comment (adfoster-r7) [reply spec/support/acceptance/command_shell.rb:1]: Or if we can't rename the existing one, can we DRY this out

File: test/modules/post/test/cmd_exec.rb [hunk 2]
```diff
@@ -155,12 +176,274 @@ def test_cmd_exec_stderr
...
+          output.rstrip == test_string
+        end
       else
         output = cmd_exec("echo #{test_string} 1>&2")
         output == test_string
       end
     end
   end
+
+  # TODO: This can be added back in once Smashery's create process API has been landed
```
Comment (adfoster-r7) [inline test/modules/post/test/cmd_exec.rb:194]: Not a blocker for this PR; but it might be easier for the next dev to pick up the code fixes here if we're wiring up this to use: `valid_show_args_response`

File: .github/workflows/meterpreter_acceptance.yml [hunk 3]
```diff
@@ -132,12 +132,11 @@ jobs:
         run: |
           REM pcap dependencies
           powershell -Command "[System.Net.ServicePointManager]::ServerCertificateValidationCallback = {$true} ; [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12; (New-Object System.Net.WebClient).DownloadFile('https://www.winpcap.org/install/bin/WpdPack_4_1_2.zip', 'C:\Windows\Temp\WpdPack_4_1_2.zip')"
```
Comment (smcintyre-r7) [inline .github/workflows/meterpreter_acceptance.yml:134]: @dwelch-r7 You mentioned we're adjusting our WpdPack dependency handling.

File: spec/support/acceptance/command_shell/linux.rb [hunk 4]
```diff
@@ -0,0 +1,158 @@
+module Acceptance::CommandShell
+  LINUX = {
+    payloads: [
+      {
+        name: "cmd/unix/reverse_bash",
+        extension: "",
```
Comment (smcintyre-r7) [inline spec/support/acceptance/command_shell/linux.rb:6]:
```suggestion
        extension: ".sh",
```

File: spec/support/acceptance/command_shell/linux.rb [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        platforms: [:linux],
+        executable: true,
+        execute_cmd: ["${payload_path}"],
```
Comment (smcintyre-r7) [inline spec/support/acceptance/command_shell/linux.rb:9]: Would you mind double checking that this fails when invoked through the shell like

```
        execute_cmd: ["sh ${payload_path}"],
```

File: spec/support/acceptance/command_shell.rb (hunk 1 above)
Comment (cgranleese-r7) [reply spec/support/acceptance/command_shell.rb:1]: Working on a prerequisite [PR](https://github.com/rapid7/metasploit-framework/pull/19428) to do some refactors to make this possible 👍

**Comment by cgranleese-r7** [comment]
I rewrote the original tests we had without having Smashery's changes to test them. Just going to leave the test that covered all the different nuances but needed improved for readability in case something was missed.

```
  # TODO: example in case `show_args_binary` needs some more massaging once we are able to test the API changes again
  #
  #  if session.platform.eql? 'windows'
  #    output = create_process('./show_args.exe', args: [test_string, '', test_string, '', test_string])
  #    if session.type.eql? 'powershell'
  #      output.rstrip == "#{pwd}\\show_args.exe\r\n#{test_string}\r\n\r\n#{test_string}\r\n\r\n#{test_string}"
  #    elsif session.type.eql? 'shell'
  #      output = create_process('show_args.exe', args: [test_string, '', test_string, '', test_string])
  #      output.rstrip == "show_args.exe\r\n#{test_string}\r\n\r\n#{test_string}\r\n\r\n#{test_string}"
  #    elsif (session.type.eql?('meterpreter') && session.arch.eql?('java'))
  #      output.rstrip == ".\\show_args.exe\r\n#{test_string}\r\n\r\n#{test_string}\r\n\r\n#{test_string}"
  #    elsif session.arch.eql?("php")
  #      # output = create_process('.\\show_args.exe', args: [test_string, '', test_string, '', test_string])
  #      # $stderr.puts output.rstrip.inspect
  #      # output.rstrip == ".\\show_args.exe\r\n#{test_string}\r\n\r\n#{test_string}\r\n\r\n#{test_string}"
  #      # TODO: Fix this functionality
  #      vprint_status("test skipped for PHP - functionality not correct")
  #      true
  #    else
  #      output.rstrip == "./show_args.exe\r\n#{test_string}\r\n\r\n#{test_string}\r\n\r\n#{test_string}"
  #    end
  #  else
  #    output = create_process('./show_args', args: [test_string, '', test_string, '', test_string])
  #    output.rstrip == "./show_args\n#{test_string}\n\n#{test_string}\n\n#{test_string}"
  #  end
```

File: spec/acceptance/command_shell_spec.rb [hunk 6]
```diff
@@ -0,0 +1,533 @@
...
+                      type: Allure::ContentType::TXT
+                    )
+
+                    current_payload_status
+                  end)
+                  expect(session_id).to_not(be_nil, proc do
+                    "There should be a session present"
+                  end)
+
+                  resource_command = "resource scripts/resource/meterpreter_compatibility.rc"
```
Comment (cgranleese-r7) [inline spec/acceptance/command_shell_spec.rb:220]: Looking at this I'm wondering should this resource script also be renamed 🤔 

It has four matches in across framework(one being a comment):
image

File: spec/support/acceptance/command_shell.rb (hunk 1 above)
Comment (cgranleese-r7) [reply spec/support/acceptance/command_shell.rb]: Got the changes from the mentioned PR pulled in now.

File: lib/msf/core/session/provider/single_command_shell.rb [hunk 7]
```diff
@@ -93,6 +93,14 @@ def shell_command_token(cmd, timeout=10)
     output
   end
 
+  def to_cmd(cmd, args)
+    if platform == 'windows'
```
Comment (adfoster-r7) [inline lib/msf/core/session/provider/single_command_shell.rb:97]: TODO: I"m not sure if this is the right fix, but if it's passing more tests - then it's definitely putting us more in the right direction :+1:

File: spec/acceptance/command_shell_spec.rb [hunk 8]
```diff
@@ -0,0 +1,533 @@
...
+                if: (
+                  # Assume that regardless of payload, staged/unstaged/etc, the command shell will have the same commands available
+                  # So only run this test when config_index == 0
+                  payload_config_index == 0 && Acceptance::Session.supported_platform?(payload_config)
+                  # Run if ENV['SESSION'] = 'java php' etc
+                  Acceptance::Session.run_session?(command_shell_config) &&
+                    # Only run payloads / tests, if the host machine can run them
+                    Acceptance::Session.supported_platform?(payload_config)
+                )
+              ) do
```
Comment (adfoster-r7) [inline spec/acceptance/command_shell_spec.rb:198]: This `it` block/test can be deleted entirely, command shells won't expose meterpreter compatibility information

File: spec/acceptance/command_shell_spec.rb (hunk 6 above)
Comment (adfoster-r7) [reply spec/acceptance/command_shell_spec.rb:220]: https://github.com/rapid7/metasploit-framework/pull/19413/files#r1768565200

File: test/modules/post/test/cmd_exec.rb [hunk 9]
```diff
@@ -21,21 +21,28 @@ def initialize(info = {})
     )
   end
 
-  def upload_precompiled_binaries
+  def upload_create_process_precompiled_binaries
```
Comment (adfoster-r7) [inline test/modules/post/test/cmd_exec.rb:24]: not a blocker: `def upload_show_args_binary`

**Review by adfoster-r7 — APPROVED** [review]
